src/transformers/models/maskformer/feature_extraction_maskformer.py

```diff
@@ -0,0 +1,565 @@
+        Resize the image to the given size. Size can be min_size (scalar) or (width, height) tuple. If size is an int,
+        smaller edge of the image will be matched to this number.
+
+        If given, also resize the target accordingly.
+        """
+        if not isinstance(image, Image.Image):
+            image = self.to_pil_image(image)
+
+        def get_size_with_aspect_ratio(image_size, size, max_size=None):
+            width, height = image_size
+            if max_size is not None:
+                min_original_size = float(min((width, height)))
+                max_original_size = float(max((width, height)))
+                if max_original_size / min_original_size * size > max_size:
+                    size = int(round(max_size * min_original_size / max_original_size))
+
+            if (width <= height and width == size) or (height <= width and height == size):
+                return (height, width)
+
+            if width < height:
+                output_width = size
+                output_height = int(size * height / width)
+            else:
+                output_height = size
+                output_width = int(size * width / height)
+
+            return (output_height, output_width)
+
+        def get_size(image_size, size, max_size=None):
+            if isinstance(size, (list, tuple)):
+                return size
+            else:
+                # size returned must be (width, height) since we use PIL to resize images
+                # so we revert the tuple
+                return get_size_with_aspect_ratio(image_size, size, max_size)[::-1]
+
+        width, height = get_size(image.size, size, max_size)
+
+        if self.size_divisibility > 0:
+            height = int(np.ceil(height / self.size_divisibility)) * self.size_divisibility
+            width = int(np.ceil(width / self.size_divisibility)) * self.size_divisibility
+
+        size = (width, height)
+        rescaled_image = self.resize(image, size=size)
+
+        has_target = target is not None
+
+        if has_target:
+            target = target.copy()
+            # store original_size
+            target["original_size"] = image.size
+            if "masks" in target:
+                masks = torch.from_numpy(target["masks"])[:, None].float()
+                #  use PyTorch as current workaround
+                # TODO replace by self.resize
+                interpolated_masks = (
+                    nn.functional.interpolate(masks, size=(height, width), mode="nearest")[:, 0] > 0.5
+                ).float()
+                target["masks"] = interpolated_masks.numpy()
+
+        return rescaled_image, target
+
+    def _normalize(self, image, mean, std, target=None):
+        """
+        Normalize the image with a certain mean and std.
+
+        If given, also normalize the target bounding boxes based on the size of the image.
+        """
+
+        image = self.normalize(image, mean=mean, std=std)
+
+        return image, target
```

Comment:
Not sure whether this method is needed, since you can just use normalize defined in image_utils.py.

Reply:
done, removed